packages/devtools_app/lib/src/screens/memory/shared/heap/model.dart

```diff
@@ -108,21 +108,28 @@ class AdaptedHeapData {
 typedef IdentityHashCode = int;
 
 /// Sequence of ids of objects in the heap.
-///
-/// TODO(polina-c): maybe we do not need to store path by objects.
-/// It can be that only classes are interesting, and we can save some
-/// performance on this object. It will become clear when the leak tracking
-/// feature stabilizes.
 class HeapPath {
   HeapPath(this.objects);
 
   final List<AdaptedHeapObject> objects;
 
+  late final bool isRetainedBySameClass = () {
+    if (objects.isEmpty) return false;
+
+    final theClass = objects.last.heapClass;
+
+    final firstWithSameClass =
```

Comment:
This is simpler and I think the intent is a bit clearer if rewritten as:

```
if (objects.length < 2) return false;
return objects.take(objects.length - 1).any((object) => object.heapClass == theClass);
````

Reply:
`take` will create one more array

Knowing that the method will run for every object, I would avoid it. 

Updated early return.

Reply:
actually, I see 'take' returns itarable

Reply:
done